code/game/mecha/mech_fabricator.dm

```diff
@@ -78,72 +75,142 @@
-			output += " | \[<a href='?src=[UID()];remove_mat=50;material=[mat_id]'>All</a>\]</span>"
-		output += "<br/>"
+	data["screen"] = screen
+	data["total_amount"] = materials.total_amount
+	data["max_amount"] = materials.max_amount
+	data["wait_message"] = wait_message
+
+	switch(screen)
+		if(MECHFAB_MAIN_MENU)
+			data["uid"] = UID()
+			data["categories"] = categories
+
+		if(MECHFAB_CATEGORY_MENU)
+			data["selected_category"] = selected_category
+			var/list/designs = list()
+			data["designs"] = designs
+			for(var/V in files.known_designs)
+				var/datum/design/D = files.known_designs[V]
+				if(D.build_type & fabricator_type && selected_category in D.category)
+					var/list/design = list()
+					designs[++designs.len] = design
+					design["name"] = D.name
+					design["id"] = D.id
+					design["materials"] = design_cost_data(D)
+			data["designs_len"] = designs.len
+
+		if(MECHFAB_SEARCH_MENU)
+			data["search"] = temp_search
+			var/list/designs = list()
+			data["designs"] = designs
+			for(var/datum/design/D in matching_designs)
+				var/list/design = list()
+				designs[++designs.len] = design
+				design["name"] = D.name
+				design["id"] = D.id
+				design["materials"] = design_cost_data(D)
+			data["designs_len"] = designs.len
+
+		if(MECHFAB_MATERIALS_MENU)
+			var/list/materials_list = list()
+			data["loaded_materials"] = materials_list
+			materials_list[++materials_list.len] = list("name" = "Metal", "id" = MAT_METAL, "amount" = materials.amount(MAT_METAL))
+			materials_list[++materials_list.len] = list("name" = "Glass", "id" = MAT_GLASS, "amount" = materials.amount(MAT_GLASS))
+			materials_list[++materials_list.len] = list("name" = "Gold", "id" = MAT_GOLD, "amount" = materials.amount(MAT_GOLD))
+			materials_list[++materials_list.len] = list("name" = "Silver", "id" = MAT_SILVER, "amount" = materials.amount(MAT_SILVER))
+			materials_list[++materials_list.len] = list("name" = "Solid Plasma", "id" = MAT_PLASMA, "amount" = materials.amount(MAT_PLASMA))
+			materials_list[++materials_list.len] = list("name" = "Uranium", "id" = MAT_URANIUM, "amount" = materials.amount(MAT_URANIUM))
+			materials_list[++materials_list.len] = list("name" = "Diamond", "id" = MAT_DIAMOND, "amount" = materials.amount(MAT_DIAMOND))
+			materials_list[++materials_list.len] = list("name" = "Bananium", "id" = MAT_BANANIUM, "amount" = materials.amount(MAT_BANANIUM))
+			materials_list[++materials_list.len] = list("name" = "Tranquillite", "id" = MAT_TRANQUILLITE, "amount" = materials.amount(MAT_TRANQUILLITE))
+			materials_list[++materials_list.len] = list("name" = "Titanium", "id" = MAT_TITANIUM, "amount" = materials.amount(MAT_TITANIUM))
+			materials_list[++materials_list.len] = list("name" = "Plastic", "id" = MAT_PLASTIC, "amount" = materials.amount(MAT_PLASTIC))
+			materials_list[++materials_list.len] = list("name" = "Bluespace Mesh", "id" = MAT_BLUESPACE, "amount" = materials.amount(MAT_BLUESPACE))

```

Comment:
Yeah I wonder if something like this would be better: 
Make a `var/list/materials_list` innate to the mech fabricator instead of locally on line 143. Loop through every material in the material datum and add each material list to `materials_list`. That way its not hardcoded.

That way when the `data["loaded_materials"] = materials_list` line executes it will just assign all the information we already assigned to `materials_list`.